src/commands/registries/azure/tasks/scheduleRunRequest.ts

```diff
@@ -169,27 +169,38 @@ async function streamLogs(context: IActionContext, node: AzureRegistryTreeItem,
     await new Promise<void>((resolve, reject) => {
         const timer = setInterval(
             async () => {
-                if (!exists && !(exists = await blobClient.exists())) {
-                    totalChecks++;
-                    if (totalChecks >= maxBlobChecks) {
-                        clearInterval(timer);
-                        reject('Not found');
+                try {
+                    if (!exists && !(exists = await blobClient.exists())) {
+                        totalChecks++;
+                        if (totalChecks >= maxBlobChecks) {
+                            clearInterval(timer);
+                            reject('Not found');
+                        }
                     }
-                }
-
-                const contentBuffer = await blobClient.downloadToBuffer(byteOffset);
-                const properties = await blobClient.getProperties();
-
-                byteOffset += contentBuffer.length;
-                const content = bufferToString(contentBuffer);
 
-                if (content) {
-                    ext.outputChannel.info(content);
-                }
+                    const properties = await blobClient.getProperties();
+                    if (properties.contentLength > byteOffset) {
+                        // New data available
+                        const response = await blobClient.download(byteOffset);
+                        byteOffset += response.contentLength;
+
+                        const lineReader = readline.createInterface(response.readableStreamBody);
+                        for await (const line of lineReader) {
+                            const sanitizedLine = line
+                                // eslint-disable-next-line no-control-regex
+                                .replace(/[\x00-\x09\x0B-\x0C\x0E-\x1F]/g, '') // Remove non-printing control characters
+                                .replace(/^\d{4}\/\d{2}\/\d{2}\s\d{2}:\d{2}:\d{2}\s/, ''); // Remove the redundant timestamps at the beginning of some lines. Their format is "YYYY/MM/DD HH:MM:SS"
```

Comment:
The `^` ensures that only the timestamps at the beginning of the lines are removed. Here's what it looks like without doing that:
![image](https://user-images.githubusercontent.com/36966225/224110230-5e6f56e9-ca52-4f92-bef5-69986617ab0b.png)

IMO it's not _that_ bad, but the inconsistency of it bugged me, since the timestamps aren't prepended to lines that are emitted by the Docker CLI--just the ones from the task runner that it is running it. I don't feel strongly about keeping it though.

Reply:
If that is the case, I think it is better to be on the safe side and not try to sanitize the output beyond just removing the non-printable chars.

Reply:
Ok, updated. GitHub will want you to approve the PR again.